src/key.rs

```diff
@@ -54,14 +80,63 @@ impl Key {
             Ok((_, ref rest)) if !rest.input.is_empty() => {
                 Err(parser::TomlError::from_unparsed(rest.positioner, s))
             }
-            Ok(((raw, key), _)) => Ok(Key::new(raw, key)),
+            Ok(((raw, parts), _)) => Ok(Key::new(raw.into(), parts)),
             Err(e) => Err(parser::TomlError::new(e, s)),
         }
     }
 
-    pub(crate) fn new(raw: &str, key: InternalString) -> Self {
+    pub(crate) fn new(raw: InternalString, parts: Vec<SimpleKey>) -> Self {
+        Self {
+            raw,
+            parts,
+        }
+    }
+
+    /// Returns the parsed key value.
+    pub fn get(&self) -> InternalString {
```

Comment:
the name `InternalString` suggests it's not for public API, we can return a `String` instead, but I'd prefer a `IntoString` impl or something like that.